src/init.cpp

```diff
@@ -1048,6 +1048,9 @@ bool AppInitParameterInteraction()
     fAcceptDatacarrier = GetBoolArg("-datacarrier", DEFAULT_ACCEPT_DATACARRIER);
     nMaxDatacarrierBytes = GetArg("-datacarriersize", nMaxDatacarrierBytes);
 
+    nLocalServices = ServiceFlags(nLocalServices | NODE_SEGWIT2X);
```

Comment:
Matching the conditionality of `NODE_WITNESS` is a slightly more delicate operation imo, wrt tests.

What used for?   Good q.   The immediate goal is visibility to DNS seeds and AddrMan peers via peer exchange.  Once this is seeded out in the field, we can (or might not need to) take further steps to discover this subset of the network.  So, this is a step in a direction, beyond seeds, where there is as yet unscoped second step re peer rendezvous.